File: src/main/java/com/dsm/pick/domains/domain/Teacher.java
package com.dsm.pick.domains.domain;

import com.dsm.pick.utils.exception.NonExistIdOrPasswordException;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Id;

@Entity
public class Teacher {

    @Id
    @Column(name = "id")
    private String id;

    @Column(name = "pw")
    private String pw;

    @Column(name = "name")
    private String name;

    @Column(name = "office")
    private String office;

    public Teacher() {}
    public Teacher(String id, String pw, String name) {
        this.id = id;
        this.pw = pw;
        this.name = name;
    }
    public Teacher(String id, String pw, String name, String office) {
        this.id = id;
        this.pw = pw;
        this.name = name;
        this.office = office;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getPw() {
        return pw;
    }

    public void setPw(String pw) {
        this.pw = pw;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getOffice() {
        return office;
    }

    public void setOffice(String office) {
        this.office = office;
    }

    public void existIdOrPassword() {
        if(this.id == null)
            throw new NonExistIdOrPasswordException();
        else if(this.pw == null)
            throw new NonExistIdOrPasswordException();
    }
}

File: src/main/kotlin/com/dsm/pick/domain/Teacher.kt
package com.dsm.pick.domain

import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.Id
import javax.persistence.Table
import javax.validation.constraints.NotBlank

@Entity
@Table(name = "teacher")
class Teacher(

    @Id
    @Column(name = "id")
    @NotBlank
    val id: String,

    @Column(name = "pw")
    @NotBlank
    var password: String,

    @Column(name = "name")
    @NotBlank
    val name: String,

    @Column(name = "office")
    @NotBlank
    val office: String,
) {

    override fun toString(): String {
        return "Teacher(id='$id', password='$password', name='$name', office='$office')"
    }
}